FreeRTOS_DNS.c

```diff
@@ -1871,6 +1871,11 @@
                         usLength = ( uint16_t ) ( sizeof( NBNSAnswer_t ) + ( size_t ) offsetof( NBNSRequest_t, usType ) );
 
                         prvReplyDNSMessage( pxNetworkBuffer, ( BaseType_t ) usLength );
+
+                        if( pxNewBuffer != NULL )
```

Comment:
Reading the most recent [post](https://forums.freertos.org/t/memory-leak-during-nbns-llmnr-hostname-resolution/12062/7) it looks like `pxNewBuffer` is out of scope by this time? Line 1823 would need to be moved up.

Reply:
I agree! Missed that completely.
@htibosch, would you mind making the requested changes?

Reply:
Sorry yes, you are right. I moved the declaration of `pxNewBuffer` upward and initialised it to NULL. Thanks